### ISSUE TITLE
Allow authors to suppress mixin output

### ISSUE THREAD

**Issue by carolynvs** [deislabs/porter#940]
Sometimes a mixin may print sensitive things that weren't already parameters or credentials. People will fall back to using the exec mixin and bash, sending output to /dev/null if we don't have a nice way to suppress mixin output natively.

```yaml
mymixin:
  description: "Do sensitive stuff"
  suppress-output: true
```

I'm not sure if `suppress-output` is the best field we can come up with. Suggestions more than welcome.

The change should be made here:

https://github.com/deislabs/porter/blob/6da074e0114723c050ce84b6025cba883718f702/pkg/exec/builder/execute.go#L80-L83

And controlled by a new custom interface like `HasCustomDashes`, so that mixins can optionally implement it by having the `suppress-output` field implemented:

https://github.com/deislabs/porter/blob/6da074e0114723c050ce84b6025cba883718f702/pkg/exec/builder/execute.go#L28-L30

Then in this PR we should implement it on on the exec mixin:

https://github.com/deislabs/porter/blob/6da074e0114723c050ce84b6025cba883718f702/pkg/exec/action.go#L47-L53

In follow-up PRs we should implement it on the other mixins like az, gcloud and aws, which all have commands that can output sensitive values (like login).

TODO:
 - [x] Main PR in deislabs/porter: https://github.com/deislabs/porter/pull/956
 - [ ] PRs for other mixins
    - [x] az: https://github.com/deislabs/porter-az/pull/13
    - [x] gcloud: https://github.com/deislabs/porter-gcloud/pull/13
    - [x] aws: https://github.com/deislabs/porter-aws/pull/16
    - [ ] other?

**Comment by vdice** [comment]
Do we wish to log a message for each step with suppressed output?  Something like `(Output suppressed)`?  Or keep it simple and simply log nothing?

**Comment by carolynvs** [comment]
How about in debug mode we print that output was suppressed (to help with troubleshooting in case someone forgot that they had it toggled) but normally we don't print anything just like the flag indicates.

**Comment by vdice** [comment]
Closed by #956 (and co.)
